pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -31,7 +30,9 @@ const OriginOutbound = "outbound"
 type OutboundProxyGenerator struct{}
 
 func (g OutboundProxyGenerator) Generate(ctx context.Context, _ *model.ResourceSet, xdsCtx xds_context.Context, proxy *model.Proxy) (*model.ResourceSet, error) {
-	hasMeshRoutes := len(proxy.Policies.Dynamic[v1alpha1.MeshHTTPRouteType].ToRules.Rules) > 0
+	if len(proxy.Routing.TrafficRoutes) == 0 {
```

Comment:
We can use routing because we set `proxy.Policies.TrafficRoutes` based on the routing https://github.com/kumahq/kuma/blob/master/pkg/xds/sync/dataplane_proxy_builder.go#L45

Reply:
Is this `> 0` exactly when there is _any_ `TrafficRoutes` in the mesh, not just ones that affect this `Dataplane`?

Reply:
I think it's about this DP. Should we validate if there is any TrafficRoute in the mesh?